test-tests/shared/src/test/scala/zio/test/GenSpec.scala

```diff
@@ -0,0 +1,64 @@
+package zio.test
+
+import zio.random.Random
+import zio.test.Assertion._
+import zio.test.TestAspect._
+import zio.ZIO
+
+object GenSpec extends ZIOBaseSpec {
+
+  def spec = suite("GenSpec")(
+    suite("zipWith")(
+      testM("left preservation") {
+        checkM(deterministic, deterministic) { (a, b) =>
+          for {
+            left  <- sample(a.zip(b).map(_._1))
+            right <- sample(a)
+          } yield assert(left, startsWith(right))
+        }
+      } @@ scala2Only,
+      testM("right preservation") {
+        checkM(deterministic, deterministic) { (a, b) =>
+          for {
+            left  <- sample(a.zip(b).map(_._2))
+            right <- sample(b)
+          } yield assert(left, startsWith(right))
+        }
+      } @@ scala2Only,
```

Comment:
Not dotty?

Reply:
Blocked by: https://github.com/zio/zio/issues/2232.